app/assets/javascripts/answers.js

```diff
@@ -35,26 +53,75 @@ enableSubmitAnswer();
 $('#answer_text').keyup(enableSubmitAnswer); 
 
 // When a user submits an answer. The real answer and response form fade in. 
-  $("#submit_answer").click(function(ev){
+$("#submit_answer").click(function(ev){
+    // prevents bootstrap from automatically adding a hidden attribute; we are explicitly hiding and showing using jquery instead
+     ev.preventDefault(); 
+     $("#answers").fadeIn();
+     $("#submit_answer").hide("slow");
+     $("#response").fadeIn();
+});
+
+$('#respond-yes').click(function(ev) {
+  $('#respond-yes').fadeIn();
+  ev.preventDefault(); // prevents bootstrap from automatically adding a hidden attribute
+  provide_feedback("yes", "Great! Congrats! Try another question.", $('#respond-yes').data('q_id'));
+  $('#response').hide("slow");
+  $('#alert_text_container').addClass("alert-success");
```

Comment:
It would make more sense to me for these classes to be added within the *provide_feedback functions themselves -- that is where we're handling the text of the alert and the showing of the alert, so determining what color the alert box should be makes sense there instead of out here in the calling function, which handles the response buttons (not the alerting portion). Separation of concerns / compartmentalization of functions and actions. Also I noticed that for the modal window version, since this code does not include any `removeClass()` calls, once you've answered a couple of questions in the review list 'yes' and 'no' then all of the alerts are red, no matter whether you answer 'yes' or 'no'. This is because at that point the `#alert_text_container` contains both `alert-success` _and_ `alert-danger` (a consequence of the fact we're not reloading the page in between calls to the modal). So we'll need something like this for the modal:
```
$('#alert_text_container').removeClass("alert-danger").addClass("alert-success");
```